### PR TITLE
Add raweth data length header field

### PR DESCRIPTION
**Pull request by jean-roland**
Because ethernet frame minimum size is 64 bytes, some device will pad frames. This forces us to add a field containing our actual payload length.

### CODE REVIEW

File: include/zenoh-pico/protocol/definitions/transport.h [hunk 1]
```diff
@@ -477,6 +477,8 @@ typedef struct {
 } _z_t_msg_fragment_t;
 void _z_t_msg_fragment_clear(_z_t_msg_fragment_t *msg);
 
+#define _Z_FRAGMENT_HEADER_SIZE 12
```
Comment (Mallets) [inline include/zenoh-pico/protocol/definitions/transport.h:480]: This is actually not true, the fragment header size may vary depending on the used protocol extension.
As a consequence, basing the code logic on this value may lead to some bugs.

File: include/zenoh-pico/system/link/raweth.h [hunk 2]
```diff
@@ -44,6 +45,7 @@ typedef struct {
     uint16_t vlan_type;                 // Vlan ethtype
     uint16_t tag;                       // Vlan tag
     uint16_t ethtype;                   // Ethertype of frame
```
Comment (Mallets) [inline include/zenoh-pico/system/link/raweth.h:47]: Do we need `ethtype` then? As far as I understand it, `data_legnth` replaces `ethtype`.

File: src/transport/multicast/tx.c [hunk 3]
```diff
@@ -116,7 +116,7 @@ int8_t _z_multicast_send_n_msg(_z_session_t *zn, const _z_network_message_t *n_m
             } else {
                 // The message does not fit in the current batch, let's fragment it
                 // Create an expandable wbuf for fragmentation
-                _z_wbuf_t fbf = _z_wbuf_make(ztm->_wbuf._capacity - 12, true);
+                _z_wbuf_t fbf = _z_wbuf_make(ztm->_wbuf._capacity - _Z_FRAGMENT_HEADER_SIZE, true);
```
Comment (Mallets) [inline src/transport/multicast/tx.c:119]: The fragmentation buffer should allocate a small amount of memory and it shouldn't be dependent on the fragment header size.

File: src/transport/unicast/tx.c [hunk 4]
```diff
@@ -125,7 +125,7 @@ int8_t _z_unicast_send_n_msg(_z_session_t *zn, const _z_network_message_t *n_msg
             } else {
                 // The message does not fit in the current batch, let's fragment it
                 // Create an expandable wbuf for fragmentation
-                _z_wbuf_t fbf = _z_wbuf_make(ztu->_wbuf._capacity - 12, true);
+                _z_wbuf_t fbf = _z_wbuf_make(ztu->_wbuf._capacity - _Z_FRAGMENT_HEADER_SIZE, true);
```
Comment (Mallets) [inline src/transport/unicast/tx.c:128]: See my comment on the fragment header size above.

File: include/zenoh-pico/protocol/definitions/transport.h (hunk 1 above)
Comment (jean-roland) [reply include/zenoh-pico/protocol/definitions/transport.h:480]: I'd say we should fix this in another PR, I opened an issue on the overall issue: #295

File: include/zenoh-pico/system/link/raweth.h (hunk 2 above)
Comment (jean-roland) [reply include/zenoh-pico/system/link/raweth.h:47]: 802.3 raw requires us to use a 802.2 LLC header which would be 3 bytes of useless information for us, so I figured an ethertype and a frame length was the most efficient choice.

**Review by Mallets — APPROVED** [review]

**Pull request merged**